src/legacy/core_plugins/kibana/public/management/sections/index_patterns/edit_index_pattern/edit_index_pattern.js

```diff
@@ -187,10 +256,33 @@ uiModules
     Promise,
     config,
     Private,
-    AppState,
     confirmModal
   ) {
-    const $state = ($scope.state = new AppState());
+    const {
+      startSyncingState,
+      stopSyncingState,
+      setCurrentTab,
+      getCurrentTab,
+      state$,
+    } = createEditIndexPatternPageStateContainer({
+      useHashedUrl: config.get('state:storeInSessionStorage'),
+    });
+
+    $scope.getCurrentTab = getCurrentTab;
+    $scope.setCurrentTab = setCurrentTab;
+
+    const stateChangedSub = state$.subscribe(({ tab }) => {
```

Comment:
If you have digest cycle issues, use `subscribeWithScope`

Reply:
thanks!